pkg/extension/sumologicextension/extension.go

```diff
@@ -200,104 +168,146 @@ func (se *SumologicExtension) Start(ctx context.Context, host component.Host) er
 	return nil
 }
 
-// handleCredentials retrieves the credentials for the collector.
-// It does so by checking the local credentials store and by validating those credentials.
-// In case they are invalid or are not available through local credentials store
-// then it tries to register the collector using the provided access keys.
-func (se *SumologicExtension) handleCredentials(ctx context.Context, host component.Host) (credentials.CollectorCredentials, bool, error) {
-	colCreds, registrationDone, err := se.getCredentials(ctx)
-	if err != nil {
-		return credentials.CollectorCredentials{},
-			registrationDone,
-			err
+// Shutdown is invoked during service shutdown.
+func (se *SumologicExtension) Shutdown(ctx context.Context) error {
+	se.closeOnce.Do(func() { close(se.closeChan) })
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+		return nil
 	}
+}
 
+func (se *SumologicExtension) validateCredentials(
+	ctx context.Context,
+) error {
+	se.logger.Info("Validating collector credentials...",
+		zap.String(collectorIdField, se.registrationInfo.CollectorCredentialId),
+		zap.String(collectorNameField, se.registrationInfo.CollectorName),
+	)
+
+	return se.sendHeartbeat(ctx)
```

Comment:
Well we could but this really checks how the credentials were injected into the http client's transport 🤔 So however we do this it will be implicit anyway (so some extent).

I'm open to suggestions.

Reply:
I think what it *should* check is whether the generated transport lets us do a valid request. Something like:
```go
httpClient := se.getHttpClient(creds)
se.sendHeartbeatWithHttpClient(httpClient)
```
and use these functions for normal heartbeats as well.